r-data-manipulation.Rmd

```diff
@@ -0,0 +1,1111 @@
+```{r}
+dogs_by_nta %>% 
+  anti_join(nta_pop, by = c("neighborhood_tabulation_area" = "nta_code")) 
+```
+
+But there are a few NTAs in the population data, with no licensed dogs:
+```{r}
+nta_pop %>% 
+  anti_join(dogs_by_nta, by = c("nta_code" = "neighborhood_tabulation_area")) 
+```
+Perhaps it's not surprising that there aren't any dogs licensed in the airport or cemetery.  What about Rikers Island? You might try a web search to see if you can figure out why none of the ~12,000 residents have dogs.
+
+### Exercise
+
+```{r, eval = FALSE, include = FALSE}
+set.seed(01819)
+library(babynames)
+
+dog_counts <- dog_licenses %>% 
+  filter(animal_gender == "F", lubridate::year(animal_birth_month) == 2016) %>% 
+  group_by(animal_name) %>% 
+  summarise(n_dogs = n()) %>% 
+  rename(name = animal_name)
+
+five_baby_names <- 
+  babynames %>% 
+  filter(year == 2016, sex == "F") %>% 
+  mutate(name = toupper(name)) %>% 
+  rename(n_babies = n) %>% 
+  semi_join(dog_counts) %>% 
+  sample_n(size = 5) %>% 
+  select(name, n_babies)
+
+# datapasta::tribble_paste(five_baby_names)
+# datapasta::tribble_paste(dog_counts %>% semi_join(five_baby_names))
+```
+Consider the following two datasets, one with a count of the number of babies (human) born in 2016 for five names:
+```{r}
+babynames <- tribble(
+      ~name, ~n_babies,
+   "CLAIRE",     5207L,
+     "ZURI",      889L,
+     "SUZY",       20L,
+    "VICKI",       13L,
+  "CHELSEA",      929L
+  )
+babynames
+```
+
+And one with the number of dogs licensed in New York whose birth month is in 2016:
+```{r}
+dognames <- tribble(     
+      ~name, ~n_dogs,
+  "CHELSEA",      1L,
+   "CLAIRE",      2L,
+     "SUZY",      3L,
+    "VICKI",      1L,
+     "ZURI",      1L
+  )
+dognames
+```
+
+* Use `left_join()` to combine the datasets.
+
+* Why doesn't this join work?  Can you fix it?
+```{r}
+babynames_2 <- tribble(
+      ~name, ~n,
+   "CLAIRE",     5207L,
+     "ZURI",      889L,
+     "SUZY",       20L,
+    "VICKI",       13L,
+  "CHELSEA",      929L
+  )
+dognames_2 <- tribble(     
+      ~name,      ~n,
+  "CHELSEA",      1L,
+   "CLAIRE",      2L,
+     "SUZY",      3L,
+    "VICKI",      1L,
+     "ZURI",      1L
+  )
+left_join(babynames_2, dognames_2)
+```
+
+* Why doesn't this join work?  Can you fix it?
+```{r}
+babynames_3 <- tribble(
+      ~name, ~n_babies,
+   "Claire",     5207L,
+     "Zuri",      889L,
+     "Suzy",       20L,
+    "Vicki",       13L,
+  "Chelsea",      929L
+  )
+dognames_3 <- tribble(     
+      ~name,      ~n_dogs,
+  "CHELSEA",      1L,
+   "CLAIRE",      2L,
+     "SUZY",      3L,
+    "VICKI",      1L,
+     "ZURI",      1L
+  )
+left_join(babynames_3, dognames_3)
+```
+
+
+### Exercise
+
+Find the dog licences per capita, but now at the community district level, by joining the `dog_licenses` data with the `nyc-cd-pop` data.
+
+## How can I read my own tabular data into R?
+
+### What is tabular data?
```

Comment:
This should be the first lesson of the chapter.

Reply:
I can appreciate following the logical order of a data analysis project, e.g. import first, but I worry leading with import doesn't build enough excitement.  I.e. why would I want to get my data into R, if I'm not entirely convinced I'll be able to do useful things once I do?  

I have some similar concerns with leading with data manipulation, rather than say plotting.  

Can we table the ordering discussion until we have some of the other sections drafted?

Reply:
:+1: we'll fix in the edit